packages/mdc-top-app-bar/_mixins.scss

```diff
@@ -0,0 +1,65 @@
+//
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+@import "@material/theme/variables"; // for mdc-theme-accessible-ink-color
+@import "@material/ripple/mixins";
+@import "./variables";
+
+//
+// Public
+//
+
+@mixin mdc-top-app-bar-ink-color($color) {
+  @include mdc-theme-prop(color, $color);
+}
+
+@mixin mdc-top-app-bar-fill-color($color) {
+  @include mdc-theme-prop(background-color, $color);
+}
+
+@mixin mdc-top-app-bar-fill-color-accessible($color) {
+  @include mdc-top-app-bar-fill-color($color);
+  @include mdc-top-app-bar-ink-color(mdc-theme-accessible-ink-color($color));
+}
+
+@mixin mdc-top-app-bar-icon-ink-color($color) {
+  .mdc-top-app-bar__icon,
+  .mdc-top-app-bar__menu-icon {
+    @include mdc-theme-prop(color, $color);
```

Comment:
I would leave `mdc-states` out of the private mixin, and include it here and pass `$color`.